examples/static-site-gen/platform/src/lib.rs

```diff
@@ -240,3 +290,136 @@ pub fn strip_windows_prefix(path_buf: PathBuf) -> std::path::PathBuf {
 
     std::path::Path::new(path_str.trim_start_matches(r"\\?\")).to_path_buf()
 }
+
+
+
+fn highlight_roc_code(code: &str) -> String {
+    let locations: Vec<Loc<Token>> = roc_parse::highlight::highlight(code);
+    let mut buf: Vec<String> = Vec::new();
+    let mut previous_fixed_token: Option<Token> = None;
+    let mut previous_text: String = "".to_string();
+    let mut current_fixed_token: Option<Token> = None;
+    let mut current_offset = 0;
+
+    buf.push(String::from("<pre><samp>"));
+
+    for location in locations {
+        let current_text = String::from(&code[current_offset..location.byte_range().end]);
+        let current_token =
+            String::from(&code[location.byte_range().start..location.byte_range().end]);
+
+        current_fixed_token = roc_parse::highlight::check_for_fixed_token(&current_token);
+
+        match location.value {
+            Token::LineComment | Token::DocComment => {
+                buf.push(String::from("<span class=\"comment\">"));
+                buf.push(current_text);
```

Comment:
On the html escaping thing, there's this crate: https://docs.rs/html-escape/latest/html_escape/.

Anywhere you're calling `buf.push(...)` here with an argument that comes from the original roc source, you probably need to be calling `buf.push(html_escape::encode_text(...))` instead. Obviously leave alone the places you're intentionally writing out tags (such as the line directly above+below this one).

Reply:
Done.